ddpui/utils/taskprogress.py

```diff
@@ -14,10 +15,9 @@ def __init__(self, task_id, hashkey, expire_in_seconds: int | None = None) -> No
         self.hashkey = hashkey
         self.task_id = task_id
         self.taskprogress = []
-        self.redis = Redis()
-        # the key doesn't exist yet, can't set the expiration
-        self.expiration_set = False
-        self.expire_in_seconds = expire_in_seconds
+        self.redis = RedisClient.get_instance()
+        if expire_in_seconds:
```

Comment:
this is from an older version, could you update to the new logic please? the PR diff has a comment explaining why